### ISSUE TITLE
Add WpGraphQL support

### ISSUE THREAD

**Issue by bonakor** [sybrew/the-seo-framework#507]
Hi,
I'm seeing a lot of interest into the WpGraphQl approach (with Gatsby), and I'd love to know if you would think about adding support for it. I've seen for Yoast Seo (https://github.com/ashhitch/wp-graphql-yoast-seo)  and would be glad having it for TSF :)

**Comment by sybrew** [comment]
I assume this would merely be an API to fetch database components without generating metadata?

That can easily be achieved with a fork from the linked project 😄
Mind that once #185 is implemented, the project needs an update.

**Comment by LeBaux** [comment]
I have seen the rising popularity of WPGraphQL myself, but my honest experience is there are not many themes built with Gatsby.js "yet". Merely tech demos, conference talks etc. Despite the 2k stars on GitHub, I wonder how many real users will utilize implementation to TSF.

On the other hand, TSF was and is known to be built with forward-thinking devs in mind and looking over that repo, it does not look like an extremely complicated "feature".

Can't say if it is worth an extension itself, but for it is another "marketing tech buzzword" we can use to promote TSF.

**Comment by sybrew** [comment]
> Can't say if it is worth an **extension** itself, [...]

It is possible to detect WPGraphQL and register a type. So we can implement it via a "[compat](https://github.com/sybrew/the-seo-framework/tree/master/inc/compat)" file, so support is automatically added without the need for an extension.

```php
add_action( 'graphql_register_types', function() {
	register_graphql_object_type(...);
	register_graphql_field(...);
} );
```

**Comment by bonakor** [comment]
Nice! And would we need to register eachfield and object type or it's possible to add it all at once?

**Comment by sybrew** [comment]
Looking at the [implementation here](https://github.com/ashhitch/wp-graphql-yoast-seo/blob/master/wp-graphql-yoast-seo.php), it seems we need to loop over each taxonomy and post type, and register the fields in bulk.

**Comment by LeBaux** [comment]
BTW I messaged you on Twitter (most convenient for me from social profiles you have on your website) with some additional details. Thank you for considering helping us with the translation!

**Comment by bjoluc** [comment]
I discovered https://github.com/findcanary/wp-graphql-the-seo-framework which works perfectly fine for my use cases – I'd love to have something along these lines included natively!

**Comment by intelligence** [comment]
Agreed, something native would be amazing. I tried using Yoast for a bit, but it's just too bloated.
